server/src/main/java/org/elasticsearch/indices/store/TransportNodesListShardStoreMetaData.java

```diff
@@ -259,22 +264,45 @@ public String toString() {
 
     public static class Request extends BaseNodesRequest<Request> {
 
-        private ShardId shardId;
+        private final ShardId shardId;
+        @Nullable
+        private final String customDataPath;
 
         public Request(StreamInput in) throws IOException {
             super(in);
             shardId = new ShardId(in);
+            if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
Maybe add a version constant here instead of repeating `Version.V_8_0_0` to make it easier to backport and remove the `BwC` logic eventually?

Reply:
This is more of a personal preference. I like to avoid introducing too many of these constants, and have not had an issue with backports so far.